File: classes/NumberList.js
class NumberList
{
    static instance;

    _list = [];

    constructor() {
        if ( !!NumberList.instance ) {
            return NumberList.instance;
        }

        NumberList.instance = this;
    }

    get list() {
        return this._list;
    }

    set list(value) {
        this._list = value;
    }
}

module.exports = NumberList;

File: classes/NumberList.java
package classes;

public class NumberList {
    private static NumberList INSTANCE;
    public long[] list;

    private NumberList(int i) {
        this.list = new long[i];
    }

    public static NumberList getInstance() {
        return getInstance(0);
    }

    public static NumberList getInstance(int i) {
        if (INSTANCE == null) {
            INSTANCE = new NumberList(i);
        }

        return INSTANCE;
    }
}
